tests/test_rebase_consensus.py

```
@@ -0,0 +1,897 @@
+        mock_diff3: AsyncMock,
+        mock_config: MagicMock,
+        mock_providers: MagicMock,
+        mock_consensus: AsyncMock,
+    ) -> None:
+        mock_config.return_value = (["m1", "m2"], 0.67, {}, None)
+        mock_providers.return_value = {"m1": AsyncMock(), "m2": AsyncMock()}
+        mock_consensus.return_value = (True, Decimal("0.05"))
+
+        call_count = 0
+
+        async def side_effect(*args: str, **kwargs: object) -> MagicMock:
```

Ruff (0.16.1)

[warning] Too many return statements (7 > 6)

(PLR0911)